source/common/network/address_impl.h

```diff
@@ -144,6 +144,11 @@ class Ipv4Instance : public InstanceBase {
   // given address if not.
   static absl::Status validateProtocolSupported();
 
+  void abslHashValue(absl::HashState state) const override {
+    absl::HashState::combine(std::move(state), ip_.ipv4_.address_.sin_addr.s_addr,
```

Comment:
The abslHashValue overrides would be more obviously correct if they used the same expressions as ther coresponding operator==. (e.g. ip_.ipv4_.address() and ip_.port())

Reply:
Emm... you are right. It is fine with IPv4 and IPv6, but I have wrong operator== for pipe and internal address, let me fix it, thanks!